core/src/test/java/org/pentaho/platform/dataaccess/datasource/api/DataSourceWizardServiceTest.java

```diff
@@ -96,6 +108,37 @@ private class DataSourceWizardServiceMock extends DataSourceWizardService {
     }
   }
 
+  /** wire {@link PentahoSystem} so PentahoSystem.get(...) returns our mocks.
+   * Similar logic to {@link AnalysisServiceTest#initPlatform()}
+   * */
+  @BeforeClass
+  public static void initPlatform() throws Exception {
+
+    MicroPlatform platform = new MicroPlatform();
+
+    IDataSourceAwareMetadataDomainRepository dataSourceAwareMetadataDomainRepository = mock( IDataSourceAwareMetadataDomainRepository.class );
```

Comment:
I don't agree with the "production" code use the default constructor in DatasourceServie/MetadataService.java/DataSourceWizardService.java. This complicates unit testing and sub class constructors.

 I also, don't agree with using  `Mockito.spy()` in unit tests, tests should never mock the class under test.
 
 So, the only way for my new tests to not use the `Mockito.spy()` was the populate/configure the `PentahoSystem.get` with the appropriate mocks. Followed the same pattern in AnalysisService
 
Ideally with more time to analyze side effects, I would correct the production use of Default constructor, so in units tests, the code wouldn't have to use `Mockito.spy()` and `MicroPlatform()` to configure `PentahoSystem.get`